src/lib/pipeline/m2/blueprint.js

```diff
@@ -0,0 +1,84 @@
+import WorkerPool from '../worker/pool';
+import M2 from './';
+
+class M2Blueprint {
+
+  static cache = new Map();
+
+  static references = new Map();
+  static pendingUnload = new Set();
+  static unloaderRunning = false;
+
+  static UNLOAD_INTERVAL = 15000;
+
+  static load(rawPath) {
+    const path = rawPath.replace(/\.md(x|l)/i, '.m2').toUpperCase();
+
+    // Prevent unintended unloading.
+    if (this.pendingUnload.has(path)) {
+      this.pendingUnload.delete(path);
+    }
+
+    // Background unloader might need to be started.
+    if (!this.unloaderRunning) {
+      this.unloaderRunning = true;
+      this.backgroundUnload();
+    }
+
+    // Keep track of references.
+    let refCount = this.references.get(path) || 0;
+    ++refCount;
+    this.references.set(path, refCount);
+
+    if (!this.cache.has(path)) {
+      this.cache.set(path, WorkerPool.enqueue('M2', path).then((args) => {
+        const [data, skinData] = args;
+
+        return new M2(path, data, skinData);
+      }));
+    }
+
+    return this.cache.get(path).then((m2) => {
+      return m2.clone();
+    });
```

Comment:
Awesome to see these blueprints coming in!

At some point I was hoping that the `load` method would not automatically instantiate an M2, but rather the blueprint could build one for you:

```
M2Blueprint.load('path/to/illidan.m2').then((blueprint) => {
  return blueprint.build();
});
```

Let's discuss this on the Gitter channel, not necessary for this PR :+1: